<a href="https://colab.research.google.com/github/LujainAldujain/data-mining-project/blob/main/Phase1/Phase1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Phase 1: Data Selection & Description

## Project Goal
<!-- tala will add the Project Goal and Introduction here -->

In [2]:
import pandas as pd

url = "https://raw.githubusercontent.com/treselle-systems/customer_churn_analysis/master/WA_Fn-UseC_-Telco-Customer-Churn.csv"
df = pd.read_csv(url)

print("Total Number of Records (Rows):", len(df))

print("Total Number of Attributes (Columns):", len(df.columns))
print("\nAttribute Names:")
print(df.columns)

print("\nAttributes Data Types:")
print(df.dtypes)

Total Number of Records (Rows): 7043
Total Number of Attributes (Columns): 21

Attribute Names:
Index(['customerID', 'gender', 'SeniorCitizen', 'Partner', 'Dependents',
       'tenure', 'PhoneService', 'MultipleLines', 'InternetService',
       'OnlineSecurity', 'OnlineBackup', 'DeviceProtection', 'TechSupport',
       'StreamingTV', 'StreamingMovies', 'Contract', 'PaperlessBilling',
       'PaymentMethod', 'MonthlyCharges', 'TotalCharges', 'Churn'],
      dtype='object')

Attributes Data Types:
customerID           object
gender               object
SeniorCitizen         int64
Partner              object
Dependents           object
tenure                int64
PhoneService         object
MultipleLines        object
InternetService      object
OnlineSecurity       object
OnlineBackup         object
DeviceProtection     object
TechSupport          object
StreamingTV          object
StreamingMovies      object
Contract             object
PaperlessBilling     object
PaymentMethod        ob


#### Overview & Dimensions
* **Total Number of Records (Rows):** 7,043 instances, where each row represents an individual telecommunications customer.
* **Total Number of Attributes (Columns):** 21 features, encompassing demographic details, subscribed services, and account/contract information, along with the target class label (`Churn`).

---

#### Attributes & Data Types Breakdown

1. **Numerical Attributes (3 Features):**
   * `SeniorCitizen` (`int64`): A binary indicator (1 for senior citizens, 0 otherwise).
   * `tenure` (`int64`): Discrete numeric feature indicating the total number of months the customer has stayed with the service provider.
   * `MonthlyCharges` (`float64`): Continuous numeric feature showing the amount billed to the customer each month.

2. **Categorical / Text Attributes (18 Features):**
   * Stored under the `object` data type in Pandas.
   * **Demographic Details:** `gender`, `Partner`, `Dependents`.
   * **Subscribed Services:** `PhoneService`, `MultipleLines`, `InternetService`, `OnlineSecurity`, `OnlineBackup`, `DeviceProtection`, `TechSupport`, `StreamingTV`, `StreamingMovies`.
   * **Account & Billing Details:** `customerID` (unique identifier), `Contract`, `PaperlessBilling`, `PaymentMethod`.
   * **Target Class Label:** `Churn` (identifies whether the customer left the company: *Yes* or *No*).

---

#### Key Observations
* **Attribute Type Inconsistency:** The attribute `TotalCharges` is currently stored as an `object` (string/categorical) data type rather than a floating-point numeric type (`float64`). This occurs because newly subscribed customers (with `tenure = 0`) have blank whitespace strings (`" "`) instead of numerical zero values. This issue will be resolved during the data preprocessing phase by converting the column into a numeric type and handling the resulting missing values.

## Dataset Description – Class & Sample
### Class Attribute & Distribution

* The target class attribute is **`Churn`**, which has two possible values: **Yes** and **No**.
* **`No`** represents customers who remained with the company, while **`Yes`** represents customers who left the company.
* The dataset contains **5,174 customers with `Churn = No`** and **1,869 customers with `Churn = Yes`**.
* This shows that the dataset contains more non-churned customers than churned customers.

---

### Raw Dataset Sample

* The first **5 records** of the raw dataset are displayed to provide a sample of the original data and its structure.

In [3]:
print("Target Column (class attribute): Churn")

count_no = df[df["Churn"] == "No"].shape[0]
count_yes = df[df["Churn"] == "Yes"].shape[0]

print(f"- Customers who did Not churn (No): {count_no}")
print(f"- Customers who did churned (Yes): {count_yes}")

print("\nSample of Raw Dataset:")
print(df.head(5))

Target Column (class attribute): Churn
- Customers who did Not churn (No): 5174
- Customers who did churned (Yes): 1869

Sample of Raw Dataset:
   customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Ye